shapeshift/web
# feat: buy token changes dynamically according to token page

## Description

1- Whenever the user opens a token page (After the token get's updated as well), it dispatches an action SET_TOKEN to populate a global store in redux with the current opened asset
2- Added a conditional validation in the useTradeRoutes to check if the current asset is valid (not undefined or null), if it is in 'ethereum' chain, and ultimately if it's not ETH itself to avoid ETH <> ETH situation
3- Implemented/adjusted the useEffect functions to make the application be tightly bound, to work accordingly only when there are expected updates

## Notice


## Pull Request Type

- [x] :nail_care: New Feature (Non-breaking Change)

## Issue (if applicable)

closes issue #934

## Testing

1. Run the application 
2. Connect the Wallet
3. Pick any token in the tokens tab to see the trading asset changing accordingly

## Screenshot

https://user-images.githubusercontent.com/60587527/153696242-bfa058f8-a4e8-4ed4-b0fb-3213311f3135.mp4
